Add section on engaged time (#347)
snowplow/documentation: docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md

- Users: Aggregates session level data to a users level, `domain_userid`, outputting the table `snowplow_web_users`.
- User Mapping: Provides a mapping between user identifiers, `domain_userid` and `user_id`, outputting the table `snowplow_web_user_mapping`. This can be used for session stitching.

## Engaged vs. Absolute Time
At a page view and session level we provide two measures of time, absolute and engaged. Absolute is simple, it's the difference between the `dervied_tstamp` of the first and last (page view or page ping) events within that page view/session, however this isn't a true measure of how long a user spent engaging with your content, and engaged time is often a large predictor of a customer conversion (whatever that may be in your domain).

> Engaged time is derived based on the page pings which means if the user isn't active on your content, the time does not count. Consider a single page view example of reading an article. If part way through reading I see something I don't understand, I may open a new tab and look this up, perhaps read a Wikipedia page on it, before coming back to your site. In this case there will be a gap in my page pings in the events data. 

Reviewer: Replace “I” with “user” or “visitor” (plural helps to avoid the awkward “he/she”)

To adjust for these gaps we calculate engaged time as the time to trigger each ping (your heartbeat) times the number of pings (ignoring the first one), and add to that the time delay to the first ping (your minimum visit length). The equation is:

$$
engaged\_time=t_{heartbeat}\times (n_{distinct\_pings} -1) + t_{min\_visit\_length}
$$ 

and the below shows an example visually.

> This calculation at a session level is slightly more involved as have to calculate these per page view and account for [stray page pings](#stray-page-pings) but the underlying idea is the same.

Reviewer: ```suggestion
At a session level, this calculation is slightly more involved, as it needs to happen per page view and account for [stray page pings](#stray-page-pings), but the underlying idea is the same.
```

- Users: Aggregates session level data to a users level, `domain_userid`, outputting the table `snowplow_web_users`.
- User Mapping: Provides a mapping between user identifiers, `domain_userid` and `user_id`, outputting the table `snowplow_web_user_mapping`. This can be used for session stitching.

## Engaged vs. Absolute Time

> At a page view and session level we provide two measures of time; absolute, how long a user had the page open, and engaged, how much of that time the user was on the page. Engaged time is often a large predictor of a customer conversion, such as a purchase or a sign-up, whatever that may be in your domain.

Reviewer 2: ```suggestion
At a page view- and session-level we provide two measures of time; **absolute**, how long a user had the page open, and **engaged**, how much of that time the user was on the page. Engaged time is often a large predictor of a customer conversion, such as a purchase or a sign-up, whatever that may be in your domain.
```

Just want to make it clear that it's at both a `page view` and `session` level, and those are two distinct entities

> To calculate absolute is simple, it's the difference between the `dervied_tstamp` of the first and last (page view or page ping) events within that page view/session.

Reviewer: ```suggestion
Calculating absolute time is simple, it's the difference between the `dervied_tstamp` of the first and last (page view or page ping) events within that page view/session.
```

> Calculating absolute time is simple, it's the difference between the `dervied_tstamp` of the first and last (page view or page ping) events within that page view/session.

Reviewer 2: ```suggestion
Calculating absolute time is simple, it's the difference between the `derived_tstamp` of the first and last (page view or page ping) events within that page view/session.
```

> Engaged time is more complicated, it is derived based on the page pings which means if the user isn't active on your content the time does not count. Let's consider a single page view example of reading an article; part way through the reader may see something they don't understand, so they open a new tab and look this up, perhaps read a Wikipedia page on it, they go down a rabbit hole and 10 minutes later they make it back to your site to finish the article. In this case there will be a gap in the page pings in the events data. 

Reviewer 2: ```suggestion
The calculation for engaged time is more complicated, it is derived based on page pings which means if the user isn't active on your content, the engaged time does not increase. Let's consider a single page view example of reading an article; partway through the reader may see something they don't understand, so they open a new tab and look this up. They might stumble upon a Wikipedia page on it, they go down a rabbit hole and 10 minutes later they make it back to your site to finish the article. In this case there will be a gap for those 10 minutes in the page pings in the events data. 
```

To adjust for these gaps we calculate engaged time as the time to trigger each ping (your heartbeat) times the number of pings (ignoring the first one), and add to that the time delay to the first ping (your minimum visit length). The formula is:

$$
engaged\_time=t_{heartbeat}\times (n_{distinct\_pings} -1) + t_{min\_visit\_length}
$$ 

> and the below shows an example visually.

Reviewer 2: ```suggestion
and the below shows an example visually for a single page view.
```

> 

Reviewer 2: Is the image meant to be here? Or should the gap between the following paragraph be removed for clarity? Just seems a bit off
